### PR TITLE
Alphabetizes field variables in the debug server

### LINKED ISSUE #14: Feature Request: Alphabetize variables in the debugger
It's quite annoying searching for variables currently, having to scroll through potentially hundreds of variables to find the one you're looking for.
It looks like it's currently grouped by inheritance from the type, but that doesn't really help when you're debugging `mob/carbon/human` or something and trying to find a `mob/`-scope var.
Even then, it's not even alphabetized within the type scope:
Code_2020-12-19_10-39-01

### PR DESCRIPTION
**Pull request by ZeWaka**
Title, also keeps the 'type' variable on top in the local section.

Resolves #14

### CODE REVIEW

**Comment by willox** [comment]
Isn't `type` a field and not a local? I think that logic would belong in `object_to_variables` (which also needs sorting.)

I don't think the args should be sorted because they should already be in a reasonable order.

The simplest way to test right now is to just not use the `AUXTOOLS_DEBUG_DLL` in your DM and hard-code the path to your built debug_server.dll. Master is compatible with SpacemanDMM atm so there's nothing to worry about there.

**Comment by ZeWaka** [comment]
Code_2020-12-20_10-18-48

File: debug_server/src/server.rs [hunk 1]
```diff
@@ -423,6 +434,8 @@ impl Server {
 					vars.push(self.value_to_variable(String::from(name), &local));
 				}
 
+				vars.sort_by_key(|a| a.name.to_lowercase());
```
Comment (SpaceManiac) [inline debug_server/src/server.rs:437]: Hmm, I'm not sure if locals should be sorted. The natural order is the declaration order which is at least somewhat useful, compared to the not-useful default order of datum vars.

Comment (ZeWaka) [reply debug_server/src/server.rs:437]: I didn't know that, seems like a decent reason to keep them unsorted, especially as we don't see hundreds of variables in locals. Any preference @willox?

Comment (willox) [reply debug_server/src/server.rs:437]: I only didn't mention it because I thought space man liked them sorted. I think it's best to only sort fields.

**Pull request merged**